lib/compress/zstd_compress.c

```diff
@@ -2491,15 +2496,22 @@ static size_t ZSTD_compressBlock_targetCBlockSize(ZSTD_CCtx* zc,
                     srcSize,
                     zc->entropyWorkspace, HUF_WORKSPACE_SIZE /* statically allocated in resetCCtx */,
                     zc->bmi2);
-            if (!ZSTD_isError(cSize) && cSize != 0) {
+
+            FORWARD_IF_ERROR(cSize);
+            if (cSize == 0) {
+                /* If compressSequences didn't work, we just output a regular
+                 * uncompressed block */
+                cSize = ZSTD_noCompressBlock(dst, dstCapacity, src, srcSize, lastBlock);
+                FORWARD_IF_ERROR(cSize);
+            } else {
                 U32 const cBlockHeader24 = lastBlock + (((U32)bt_compressed)<<1) + (U32)(cSize << 3);
                 MEM_writeLE24(ostart, cBlockHeader24);
                 cSize += ZSTD_blockHeaderSize;
             }
         }
     }
 
-    if (!ZSTD_isError(cSize) && cSize != 0) {
+    if (!ZSTD_isError(cSize) && !usingNoCompressSuperBlock) {
         /* confirm repcodes and entropy tables when emitting a compressed block */
```

Comment:
OK, something I'm not completely clear here :
the comment states : 
> confirm repcodes and entropy tables when emitting a compressed block

So, obviously, when `ZSTD_isError(cSize)`, no compressed block is emitted, since nothing is generated.

However, when `(!usingNoCompressSuperBlock)`, there is still a change that a compress block was sent. It's just no longer divided into sub-blocks.

So I'm now wondering what would happen in this case, especially if a `repcode` is used in the next block.